webrtc.html

```diff
@@ -1549,81 +1549,104 @@ <h4>Set the RTCSessionDescription</h4>
+                              multiplexing is negotiated according to [[!RFC5761]],
+                              or if <var>connection</var>'s
+                              <code><a>RTCRtcpMuxPolicy</a></code> is <code><a
+                              data-link-for="RTCRtcpMuxPolicy">require</a></code>,
+                              do not create any RTCP-specific transport objects,
+                              and instead let <var>rtcpTransport</var> equal
+                              <var>transport</var>.
+                            </p>
+                          </li>
+                          <li>
+                            <p>
+                              Set <var>transceiver</var>.<a>[[\Sender]]</a>.<a>[[\SenderTransport]]</a>
+                              to <var>transport</var>.
+                            </p>
+                          </li>
+                          <li>
+                            <p>
+                              Set <var>transceiver</var>.<a>[[\Sender]]</a>.<a>[[\SenderRtcpTransport]]</a>
+                              to <var>rtcpTransport</var>.
+                            </p>
+                          </li>
+                          <li>
+                            <p>
+                              Set <var>transceiver</var>.<a>[[\Receiver]]</a>.<a>[[\ReceiverTransport]]</a>
+                              to <var>transport</var>.
+                            </p>
+                          </li>
+                          <li>
+                            <p>
+                              Set <var>transceiver</var>.<a>[[\Receiver]]</a>.<a>[[\ReceiverRtcpTransport]]</a>
+                              to <var>rtcpTransport</var>.
+                            </p>
+                          </li>
+                        </ol>
                       </li>
                       <li>
-                        <p>Set <var>transceiver</var>'s <code><a data-link-for=
-                        "RTCRtpTransceiver">mid</a></code> value to the mid of
-                        the corresponding <a>media description</a>.</p>
+                        <p>Let <var>transceiver</var> be the <code>
+                        <a>RTCRtpTransceiver</a></code> <a>associated</a> with the
+                        <a>media description</a>.</p>
                       </li>
                       <li>
                         <p>If <var>transceiver</var>'s <a>[[\Stopped]]</a> slot
                         is <code>true</code>, abort these sub steps.</p>
                       </li>
                       <li>
                         <p>If <var>description</var> is of type
-                        <code>"answer"</code> or <code>"pranswer"</code>, then
-                        set <var>transceiver</var>'s <a>[[\CurrentDirection]]</a> slot
+                        <code>"offer"</code> 
+                        set <var>transceiver</var>'s <a>[[\OfferedDirection]]</a> slot
```

Comment:
I think we also need to set `[[OfferedDirection]]` in SRD(offer). Otherwise, if packets arrive inbetween SRD(offer) and SLD(answer) then we might accidentally unmute just-muted tracks, because `[[CurrentDirection]]` isn't updated yet (it gets updated very soon in SLD(answer) but this leaves a timing-hole).

I also worry about rollback.

Reply:
I worry about rollback as well.

But I don't think we should set [[OfferedDirection]] in SRD at all. [[OfferedDirection]] just indicates that the offerer has an outstanding offer (and should be prepared to receive if the [[OfferedDirection]] says so).
The answerer doesn't need to use (or set [[OfferedDirection]]), instead it decides if it's going to accept the offered session at SLD(answer). And SLD(answer) updates [[CurrentDirection]], and before [[CurrentDirection]] says so the answerer should not accept media IIUC.

Reply:
This does not solve the problem in issue #1707 at all. This issue arises when:

answerer->SRD(offer); <--- offer that mutes a track
Stray RTP packet for track arrives at answerer, unmuting the track
answerer->SLD(answer); <--- doesn't alter the muted state of the track

I really do think the simplest/easiest/most likely to work fix is to disable  "unmute on RTP" when the signaling state isn't "stable".

Reply:
@docfaraday yes, that's an oversight by me. I just changed the PR. Would this work as intended?